### PR TITLE
Feat currency type optimistic cache

### LINKED ISSUE #3806: Currency type is not implemented in optimistic cache update
## Bug Description

- Go to /companies table page
- filter on a currency field (example ARR)
- update a record in the table
- You should see a red snack bar "Not implemented yet" appearing

## Technical inputs

The error is triggered by our optimistic update of our cache ; we are trying to update the filtered query results based on the update but the Currency field type is not supported yet
Update packages/twenty-front/src/modules/object-record/record-filter/utils/isRecordMatchingFilter.ts to support Currency type

### PR DESCRIPTION
**Pull request by anoopw3bdev**
Fixes #3806

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-filter/utils/isMatchingCurrencyFilter.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      case currencyFilter.amountMicros.is !== undefined: {
+        if (currencyFilter.amountMicros.is === 'NULL') {
+          return value === null;
+        } else {
+          return value !== null;
+        }
+      }
+      default: {
+        throw new Error(
+          `Unexpected value for currency filter : ${JSON.stringify(
```
Comment (AbuHafsa) [inline packages/twenty-front/src/modules/object-record/record-filter/utils/isMatchingCurrencyFilter.ts:42]: Thank you for your contribution first of all I just had a small suggestion to change the keyword `value` to `operation` as it makes more sense. 

On top of the unit tests, I did some manual tests and it seems to be working. @charlesBochet

Comment (charlesBochet) [reply packages/twenty-front/src/modules/object-record/record-filter/utils/isMatchingCurrencyFilter.ts:42]: Thanks for reviewing it @AbuHafsa
If we want to be accurate value should be amountMicros

Comment (anoopw3bdev) [reply packages/twenty-front/src/modules/object-record/record-filter/utils/isMatchingCurrencyFilter.ts:42]: Thanks @AbuHafsa @charlesBochet for the suggestion, I will push the changes.

**Review by charlesBochet — APPROVED** [review]
Sorry for the slow review @anoopw3bdev. It works well, looks good to me!

Thank you!

**Pull request merged**